### ISSUE TITLE
[prefer-regexp-exec] Dynamic RegEx with global flag triggers rule

### ISSUE THREAD

**Issue by TrevorSayre** [typescript-eslint/typescript-eslint#2873]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-regexp-exec": "warn"
  }
}
```

```TS
const stringPartitions = myString.match(new RegExp(`.{1,${partitionLength}}`, "g"));
```

```json
{
  "compilerOptions": {
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "baseUrl": "src",
    "esModuleInterop": true,
    "importsNotUsedAsValues": "error",
    "module": "CommonJS",
    "moduleResolution": "Node",
    "noImplicitAny": true,
    "noImplicitReturns": true,
    "noUnusedLocals": true,
    "outDir": "dist",
    "removeComments": true,
    "resolveJsonModule": true,
    "sourceMap": true,
    "strict": true,
    "target": "ES2019",
  },
  "include": ["src/**/*"],
  "exclude": ["node_modules"]
}

```

**Expected Result**

The rule states:
> `RegExp#exec` is faster than `String#match` and both work the same when not using the `/g` flag.

This does not warn for RegEx literals with the global flag, as expected:

```ts
const stringPartitions = myString.match(/.{1,10}/g);
```

**Actual Result**

But it does warn for RegEx created with the constructor with the global flag, not expected:

```ts
const stringPartitions = myString.match(new RegExp(`.{1,${partitionLength}}`, "g"));
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^4.9.0` |
| `@typescript-eslint/parser`        | `^4.9.0` |
| `TypeScript`                       | `^4.1.2` |
| `ESLint`                           | `^7.15.0` |
| `node`                             | `12.18.3` |

**Comment by DanKaplanSES** [comment]
Does https://github.com/typescript-eslint/typescript-eslint/issues/3477 duplicate this?

**Comment by rafaelss95** [comment]
I was unable to reproduce this with the current on `master`.. so unless I'm missing something, it was already fixed.

**Comment by bradzacher** [comment]
cannot repro this either, closing.
